### PR TITLE
pass annotations from scheduler to created backup

### LINKED ISSUE #2348: Reflect schedule annotations to backups for more flexibility storing metadata
**Describe the problem/challenge you have**
We need to store some metadata along with the backup like some spec indicating how we actually creating/managing that cluster. This spec includes Kubernetes version as well as versions/configs of components that we installed on the cluster.

Currently, it is possible to store some metadata with labels and labels on schedule is[ reflected to backups](https://github.com/vmware-tanzu/velero/blob/master/pkg/builder/backup_builder.go#L78-L85) and velero stores whole backup resource including the labels. This is good for storing some metadata but label length is limited to 63 chars.

**Describe the solution you'd like**
Annotations on velero schedule should also be reflected to backups created from them. This way we would have more flexibility on keeping metadata of backups.

**Anything else you would like to add:**
I would expect this to be a simple change without breaking any existing flow, hope we can have this soon :)

**Environment:**

- Velero version (use `velero version`): v1.2.0
- Kubernetes version (use `kubectl version`): v1.17.3
- Kubernetes installer & version: N/A
- Cloud provider or hardware configuration: N/A
- OS (e.g. from `/etc/os-release`): N/A

### PR DESCRIPTION
**Pull request by funkycode**
This should should pass annotations from scheduler to backup it creates

Fixes #2348

### CODE REVIEW

File: changelogs/unreleased/3067-funkycode [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Pass annotations from schedluer to backup it creates the same way it is done for labels
```
Comment (nrb) [inline changelogs/unreleased/3067-funkycode:1]: `schedluer` -> `schedule`

**Review by nrb — CHANGES_REQUESTED** [review]
Thanks for this contribution, I really appreciate it! I have one minor spelling change that I can see, but otherwise it looks good and I don't think there are any major ramifications.

File: changelogs/unreleased/3067-funkycode (hunk 1 above)
Comment (funkycode) [reply changelogs/unreleased/3067-funkycode]: done

**Review by ashish-amarnath — APPROVED** [review]
LGTM!

**Review by carlisia — APPROVED** [review]
lgtm, thank you! 👍

**Pull request merged**